src/Components/Web.JS/src/Services/NavigationEnhancement.ts

```diff
@@ -116,13 +119,26 @@ function onDocumentSubmit(event: SubmitEvent) {
       return;
     }
 
+    const submitter = event.submitter as HTMLButtonElement;
```

Comment:
Strictly speaking I'm not sure this is a valid downcast. A form could be submitted by something other than a button, e.g., you can have `<input type="image">` that triggers the form submit. Bear in mind that `as` in TypeScript just means "cast" (unlike in C#, where `as` has a different meaning because in C# it also does a runtime type check and supplies `null` if the type check is not satisfied).

Based on JS's loose typing the code would probably work anyway even in that case, but to communicate the intent more clearly, it may be clearer to avoid the downcast, and instead just do:

```ts
if (event.submitter?.getAttribute('formmethod') === 'dialog' || formElem.method === 'dialog') { ... }
```

Reply:
BTW I'm aware this downcast already existed in the code before, and was probably my mistake in writing an earlier version of this!